atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java

```diff
@@ -2428,6 +2436,29 @@ public void setsRequestedCommitLocksCountCorrectly_serializableLockLevelMigratio
         assertThat(commitLockInfo.rowCommitLocksRequested()).isEqualTo(1 + 1);
     }
 
+    @Test
+    public void exceptionThrownWhenTooManyPostFilterIterationsOccur() {
+        // Need to block deleter executor from deleting aborted values to help bloat the row.
+        doNothing().when(deleteExecutor).execute(any());
+
+        for (int idx = 0; idx < SnapshotTransaction.MAX_POST_FILTERING_ITERATIONS; idx++) {
+            try {
+                txManager.runTaskWithConditionThrowOnConflict(ALWAYS_FAILS_CONDITION, (txn, _c) -> {
+                    txn.put(TABLE_NO_SWEEP, ImmutableMap.of(TEST_CELL, TEST_VALUE));
+                    return null;
+                });
+            } catch (TransactionFailedRetriableException _t) {
+            }
```

Comment:
nit: leave a note saying this is expected, as we want the transaction to fail (for purposes of reproducing the exact situation).